runJupyterDaskOnSLURM.py

```diff
@@ -0,0 +1,268 @@
+    """
+    command line argument parser
+
+    accepted arguments:
+        mutualy exclusive, one required:
+        --add_platform (-a) : The script will query the user for login and connection information for the platform.
+                              This information is then saved in ./config/platforms/platforms.ini for future use
+        --one_off (-oo)     : As for add platform, except that the information entered is NOT saved
+        --platform (-p)     : The script will look for the login and connection information for the platform specified 
+                              by the string passed and will use this, if successful.
+
+        optional:
+        --local_port (-lp)  : The script will set up portforwarding to the specified port of the localhost. 
+                              If not specified port 8889 is used 
+
+    :return args: ArgumentParser return object containing command line arguments  
+
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--local_port", "-lp", help="specify non-default local port for port forwarding", type=str)
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--add_platform", "-a", help="add new platform and store config", action="store_true")
+    group.add_argument("--one_off", "-oo", help="one-off interactive configuration", action="store_true")
+    group.add_argument("--platform", "-p", help="Make use of configuration for known platform as saved in platforms.ini.", type=str)
+    args = parser.parse_args()
+    return args 
+
+def get_verified_input(prompt):
+    """
+    Query user for input on command line using specified prompt.
+    Register input, display input and ask for confirmation of correctness. Only proceed upon confirmation.
+
+    :param prompt: str; prompt to display to user requesting input
+    :return userinput: str; user supplied input  
+    """
+    unverified = True
+    while unverified:
+        userinput = input(prompt+'\n')
+        print(userinput+'\n')
+        verification = input('is this correct? [Y]/n]'+'\n') or 'Y'
+        print(verification+'\n')
+        if verification == 'Y':
+            unverified = False
+    return userinput
+
+def add_platform(oneoff=False):
+    """
+    Query information from user required to use a platform in the context of JupyterDaskOnSLURM. Optionally save.
+    Returns input information.
+    Queries:
+        platform name
+        host alias, e.g. USER@spider.surf.nl, where spider.surf.nl is the host alias
+        user name for platform
+        absolute (local) path to SSH key granting access to platform
+
+    :param oneoff: default False; if true DO NOT save input to config file, but only use for this deployment
+    :return config_inputs: dictionary with fields {host: user: keypath:}
+    :return platform_name: name specifying platform (is used in selecting job scripts) 
+    
+    """
+    platform_name = get_verified_input('Please enter platform name:')
```

Comment:
Right now this can only be `spider` or `snellius`, right? Maybe we should specify it?

Reply:
In principle this can be any platform that JupyterDaskOnLURM has been installed on though, right? It would be up to the user to adapt the job submission script, but that is stated in the README and implied in the inline documentation. The first invocation must ue either the -a or -oo arguments, and creates the relevant entry in the configuration file.

Reply:
Yes, I agree this could be used anywhere, but I think at the moment "platform" is what is used to choose the SLURM batch job?
```
jscript = 'jupyter_dask_'+platform+'.bsh'
``` 
Maybe we should get rid of the two versions of the script, use a single `jupyter_dask.bsh` file and clearly state in the REAMDE file where things have to be modified in order to start this on Spider or on Snellius?

Reply:
I've left this for now. Until we supply some support for generating these scripts I'd prefer to keep them as templates rather than having to add (a lot of) text in the documentation.